### PR TITLE
Adding ability for a BSL with credentials to pass validation.

### LINKED ISSUE #475: Bug: velero.credential.name does't take effect
### Contact Details

xjiang@redhat.com

### Describe bug

OADP version: 0.5.0

When I try to use GCP provider, the configuration as below, the credential name `cloud-credentials` does't take effect. it still tried to look for Secret `cloud-credentials-gcp`. 

```
apiVersion: oadp.openshift.io/v1alpha1
kind: DataProtectionApplication
metadata:
  name: dpa-sample
  namespace: openshift-adp
spec:
  backupLocations:
    - velero:
        credential:
          key: cloud
          name: cloud-credentials
        objectStorage:
          bucket: oadptest
          prefix: velero
        default: true
        provider: gcp
  configuration:
    restic:
      enable: true
    velero:
      defaultPlugins:
        - gcp
        - csi
  snapshotLocations:
    - velero:
        config:
          project: openshift-qe
          snapshotLocation: us-west1
        provider: gcp
        default: true
```

The command used to create secret as below:
`$ oc create secret generic cloud-credentials --namespace openshift-adp --from-file cloud=credentials-velero-gcp`

Error:
```
  spec:
    backupLocations:
    - velero:
        config:
          profile: default
        credential:
          key: cloud
          name: cloud-credentials
        default: true
        objectStorage:
          bucket: oadptest
          prefix: velero
        provider: gcp
    configuration:
      restic:
        enable: true
      velero:
        defaultPlugins:
        - gcp
        - csi
    snapshotLocations:
    - velero:
        config:
          profile: default
          region: us-west1
        provider: gcp
  status:
    conditions:
    - lastTransitionTime: "2021-11-25T08:09:54Z"
      message: Secret "cloud-credentials-gcp" not found
      reason: Error
      status: "False"
      type: Reconciled
```

### What happened?

A bug happened!

### OADP Version

0.4.x (Beta)

### OpenShift Version

4.9

### Velero pod logs

_No response_

### Restic pod logs

_No response_

### Operator pod logs

_No response_

### New issue

- [X] This issue is new

### PR DESCRIPTION
**Pull request by shawn-hurley**
fixes #475

### CODE REVIEW

**Review by dymurray — APPROVED** [review]

File: controllers/velero.go [hunk 1]
```diff
@@ -629,6 +629,11 @@ func (r DPAReconciler) noDefaultCredentials(dpa oadpv1alpha1.DataProtectionAppli
 		if bsl.Velero != nil && bsl.Velero.Credential == nil {
```
Comment (kaovilai) [inline controllers/velero.go:629]:
```suggestion
		bslProvider := strings.TrimPrefix(bsl.Velero.Provider, "velero.io/")
		if bsl.Velero != nil && bsl.Velero.Credential == nil {
```

File: controllers/velero.go [hunk 2]
```diff
@@ -629,6 +629,11 @@ func (r DPAReconciler) noDefaultCredentials(dpa oadpv1alpha1.DataProtectionAppli
...
 			providerNeedsDefaultCreds[strings.TrimPrefix(bsl.Velero.Provider, "velero.io/")] = true
```
Comment (kaovilai) [inline controllers/velero.go:630]:
```suggestion
			providerNeedsDefaultCreds[bslProvider] = true
```

File: controllers/velero.go [hunk 3]
```diff
@@ -629,6 +629,11 @@ func (r DPAReconciler) noDefaultCredentials(dpa oadpv1alpha1.DataProtectionAppli
...
 		}
+		if bsl.Velero != nil && bsl.Velero.Credential != nil {
+			if found := providerNeedsDefaultCreds[strings.TrimPrefix(bsl.Velero.Provider, "velero.io/")]; !found {
```
Comment (kaovilai) [inline controllers/velero.go:633]:
```suggestion
			if found := providerNeedsDefaultCreds[bslProvider]; !found {
```

File: controllers/velero.go [hunk 4]
```diff
@@ -629,6 +629,11 @@ func (r DPAReconciler) noDefaultCredentials(dpa oadpv1alpha1.DataProtectionAppli
...
+				providerNeedsDefaultCreds[strings.TrimPrefix(bsl.Velero.Provider, "velero.io/")] = false
```
Comment (kaovilai) [inline controllers/velero.go:634]:
```suggestion
				providerNeedsDefaultCreds[bslProvider] = false
```

**Comment by shawn-hurley** [comment]
+1 on the suggestions @kaovilai

**Pull request merged**